### PR TITLE
CI: replace `lintr` by `styler` and fix all R code

### PR DESCRIPTION
**Pull request by bernt-matthias**
`styler` is the R code linter/styler from tidyverse. in contrast to `lintr` it can also fix the found problems (added a script that can be used for checking and fixing). Main advantage is that we can apply this script once for all R scripts and do not have to bother contributors with errors that they are not responsible for.

In the testbump https://github.com/galaxyproject/tools-iuc/pull/5683/commits/1947881cce28c29f8e037a54355c9feacac394ed one can see it in action https://github.com/galaxyproject/tools-iuc/actions/runs/7225652632/job/19689585872?pr=5683

I would suggest **[skip-ci]**, i.e. not bump or deploy the changed tools.
There are two linter errors left (URLs: for egsea I contacted the website admins, mageck will be fixed [here](https://github.com/galaxyproject/tools-iuc/pull/5684)) which should be fine if we agree on skip-ci.

Do we want to run the lintr job also in weekly CI? 

If we want to pin the version of the installed packages (i.e. of styler) we need to use (since setup-r-dependencies does not really support setting the version)
```
    - name: Install packages
      run: |
        R -e 'install.packages("argparse", version="X.Y.Z", repos="http://cran.rstudio.com/")'
        R -e 'install.packages("styler", version="A.B.C", repos="http://cran.rstudio.com/")'
```

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [x] - This PR does something else (explain below)

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
This is scary, but certainly a big simplification for everyone.

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -202,38 +202,20 @@ jobs:
...
-        remotes::install_cran("lintr")
-      shell: Rscript {0}
+    - name: Install packages
+      uses: r-lib/actions/setup-r-dependencies@v2
+      with:
+        packages: |
+          any::argparse 
+          any::styler
     - name: Save repositories to file
       run: echo '${{ needs.setup.outputs.repository-list }}' > repository_list.txt
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:212]:
```suggestion
```

**Comment by bernt-matthias** [comment]
> This is scary

May I ask in which respect?

**Comment by mvdbeek** [comment]
I don't know at what level the fixer works, and I don't know if it is guaranteed to be correct.

**Comment by bernt-matthias** [comment]
> I don't know at what level the fixer works, and I don't know if it is guaranteed to be correct.

I also can't answer these questions. At least it promises to be "non-invasive" and I and colleagues are using it for some while without problems. 

Still we will have to review the R code .. anyway.

**Comment by bernt-matthias** [comment]
What's your opinion regarding bumping/not? Depending on this I will factor out the changes to multigsea to a separate PR.

**Comment by mvdbeek** [comment]
I would not bump or deploy, this should hopefully not have any effect ?

**Comment by bernt-matthias** [comment]
OK. I reverted the changes to mutigsea and opened a separate PR. So, now the PR only contains changes to the R code (made by `styler`) and 2 or 3 fixes to URLs.

Ready from my side.

**Comment by bernt-matthias** [comment]
> Ready from my side. 

Remaining question is only if we want to add this to weekly CI or pin the styler version. Guess we should do one of these two options.

**Comment by bernt-matthias** [comment]
Could we merge this? Or should I do anything here?

**Review by bgruening — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks a lot @bernt-matthias!

**Pull request merged**